File: leetcode/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/solution.c
int minOperations(int* nums, int n) {
    int f[n] = {};
    int ret = 0;
    for(int i = 0;i<n-2;i++) {
        if ((nums[i] + f[i])%2) continue;
        f[i] += 1;
        f[i+1] += 1;
        f[i+2] += 1;
        ret += 1;
    }
    if ((f[n-1] + nums[n-1])%2 && 
        (f[n-2] + nums[n-2])%2) {
        return ret;
    }
    return -1;
}

File: leetcode/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/solution.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut f = vec![0;n];
        let mut ret = 0;
        for i in 0..n-2 {
            if (nums[i] + f[i])%2 == 0 {
                f[i] += 1;
                f[i+1] += 1;
                f[i+2] += 1;
                ret += 1;
            }
        }
        if (f[n-1] + nums[n-1])%2 == 1 && (f[n-2] + nums[n-2])%2 == 1 {
            return ret;
        }
        return -1;
    }
}
